src/test/java/org/mifos/integrationtest/cucumber/stepdef/BillPayStepDef.java

```diff
@@ -309,4 +323,195 @@ public void iCanExtractTheCallbackBodyAndAssertTheRtpStatus() {
         }
         assertThat(flag).isTrue();
     }
+
+    @Then("I should be able to extract response body from callback for biller unidentified")
+    public void iShouldBeAbleToExtractResponseBodyFromCallbackForBillerUnidentified() {
+        boolean flag = false;
+        List<ServeEvent> allServeEvents = getAllServeEvents();
+        for (int i = allServeEvents.size() - 1; i >= 0; i--) {
+            ServeEvent request = allServeEvents.get(i);
+            if (!(request.getRequest().getBodyAsString()).isEmpty()) {
+                JsonNode rootNode = null;
+                flag = true;
+                try {
+                    rootNode = objectMapper.readTree(request.getRequest().getBody());
+                    logger.info("Rootnode value:" + rootNode);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+                if (rootNode != null && rootNode.has("responseCode") && rootNode.get("responseCode").asText().equals("01")) {
+                    String responseDescription = null;
+                    if (rootNode.has("responseDescription")) {
+                        responseDescription = rootNode.get("responseDescription").asText();
+                    }
+                    assertThat(responseDescription).isNotEmpty();
+                    assertThat(responseDescription).contains("Unindentified Biller");
+                    String responseCode = null;
+                    if (rootNode.has("responseCode")) {
+                        responseCode = rootNode.get("responseCode").asText();
+                    }
+                    assertThat(responseCode).isNotEmpty();
+                }
+            }
+
+        }
+        assertThat(flag).isTrue();
+    }
+
+    @Then("I should be able to extract response body from callback for bill invalid")
+    public void iShouldBeAbleToExtractResponseBodyFromCallbackForBillInvalid() {
+        boolean flag = false;
+        List<ServeEvent> allServeEvents = getAllServeEvents();
+        for (int i = allServeEvents.size() - 1; i >= 0; i--) {
+            ServeEvent request = allServeEvents.get(i);
```

Comment:
Looks like this code is repeated all over, can we use a single method for finding in events and reuse them?

Reply:
i tried but since the repeated code has a loop and multiple values getting assigned asserts are failing either due to not being able to run the loop properly or loss of a value from method. I tried creating an custom object of sorts and  use that to return the value but the parsing and extraction of values is becoming a problem.

Reply:
https://fynarfin.atlassian.net/browse/PHEE-559